chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/validation/reviewer/TestTaskReviewer.java

```diff
@@ -0,0 +1,414 @@
+        } else if (testTask.getTaskTimeDeviationOptimalAvg() != null) {
+            try {
+                Integer.valueOf(testTask.getTaskTimeDeviationOptimalAvgStr());
+            } catch (NumberFormatException ex) {
+                try {
+                    int val = Math.round(Float.valueOf(testTask.getTaskTimeDeviationOptimalAvgStr()));
+                    listing.getWarningMessages().add(
+                            msgUtil.getMessage("listing.criteria.roundedTestTaskNumber",
+                                    formatTaskRef(testTask), "Task Time Deviation Optimal Average",
+                                    testTask.getTaskTimeDeviationOptimalAvgStr(), String.valueOf(val)));
+                } catch (Exception ignore) {
+                }
+            }
+        }
+    }
+
+    private void reviewTestTaskErrors(CertifiedProductSearchDetails listing, TestTask testTask) {
+        if (testTask.getTaskErrors() == null && !StringUtils.isEmpty(testTask.getTaskErrorsStr())) {
+            listing.getErrorMessages().add(
+                    msgUtil.getMessage("listing.criteria.invalidTestTaskErrors", formatTaskRef(testTask), formatTaskCriteria(testTask),
+                            testTask.getTaskErrorsStr()));
+        } else if (testTask.getTaskErrors() == null && StringUtils.isEmpty(testTask.getTaskErrorsStr())) {
+            listing.getErrorMessages().add(
+                    msgUtil.getMessage("listing.criteria.missingTestTaskErrors", formatTaskRef(testTask), formatTaskCriteria(testTask)));
+        }
+    }
+
+    private void reviewTestTaskErrorsStddev(CertifiedProductSearchDetails listing, TestTask testTask) {
+        if (testTask.getTaskErrorsStddev() == null && !StringUtils.isEmpty(testTask.getTaskErrorsStddevStr())) {
+            listing.getErrorMessages().add(
+                    msgUtil.getMessage("listing.criteria.invalidTestTaskErrorsStddev", formatTaskRef(testTask), formatTaskCriteria(testTask),
+                            testTask.getTaskErrorsStddevStr()));
+        } else if (testTask.getTaskErrorsStddev() == null && StringUtils.isEmpty(testTask.getTaskErrorsStddevStr())) {
+            listing.getErrorMessages().add(
+                    msgUtil.getMessage("listing.criteria.missingTestTaskErrorsStddev", formatTaskRef(testTask), formatTaskCriteria(testTask)));
+        }
+    }
+
+    private void reviewTestTaskRatingScale(CertifiedProductSearchDetails listing, TestTask testTask) {
+        if (StringUtils.isEmpty(testTask.getTaskRatingScale())) {
+            listing.getErrorMessages().add(msgUtil.getMessage("listing.criteria.missingTestTaskRatingScale",
+                    formatTaskRef(testTask), formatTaskCriteria(testTask)));
+        }
+        //TODO: should we add validation to confirm it is in a set of allowed values? This was not done before as best I can tell
```

Comment:
Do you want to put a checklist item on the ticket to track this question? It might be useful, but could also open a can of worms

Maybe worth asking about those Test Participants who have "number of months of experience" values set to the year they started, so they show 2012 or whatever in the data

Reply:
Added checklist item with this question